contracts/interfaces/ICover.sol

```diff
@@ -179,7 +185,7 @@ interface ICover {
   event StakingPoolCreated(address stakingPoolAddress, address manager, address stakingPoolImplementation);
   event ProductTypeUpserted(uint id, string ipfsMetadata);
   event ProductUpserted(uint id, string ipfsMetadata);
-  event CoverBought(uint coverId, uint productId, uint segmentId, address buyer);
+  event CoverBought(uint coverId, uint productId, uint segmentId, address buyer, string ipfsData);
```

Comment:
You could name this `ipfsMetadata` as above since it's data about the on-chain data, not usable in contracts.